pcdsdevices/lens.py

```diff
@@ -29,15 +29,68 @@ def __init__(self, prefix, *, name, **kwargs):
             self._transmission[state] = self._lens_transmission
         super().__init__(prefix, name=name, **kwargs)
 
-
-# Change into PseudoPositioner when it's time to add the calculations
-class LensStack(Device):
+class LensStack(PseudoPositioner):
     x = FCpt(IMS, '{self.x_prefix}')
     y = FCpt(IMS, '{self.y_prefix}')
     z = FCpt(IMS, '{self.z_prefix}')
+    
+    calib_z = Cpt(PseudoSingle)
 
     def __init__(self, x_prefix, y_prefix, z_prefix, *args, **kwargs):
         self.x_prefix = x_prefix
         self.y_prefix = y_prefix
         self.z_prefix = z_prefix
         super().__init__(x_prefix, *args, **kwargs)
+
+    def tweak(self):
+        """
+        Calls the tweak function from mv_interface
+        with the x and y motors.
+        """
+        tweak_base(self.x,self.y)
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        z_pos = pseudo_pos.calib_z
+        setup_preset_paths(hutch='presets',exp='presets')
+        pos = [self.x.presets.positions.entry.pos,
+               self.y.presets.positions.entry.pos,
+               self.z.presets.positions.entry.pos,
+               self.x.presets.positions.exit.pos,
+               self.y.presets.positions.exit.pos,
+               self.z.presets.positions.exit.pos]
+        x_pos = ((pos[0]-pos[3])/(pos[2]-pos[5]))*(z_pos-pos[2])+pos[0]
+        y_pos = ((pos[1]-pos[4])/(pos[2]-pos[5]))*(z_pos-pos[2])+pos[1]
+        return self.RealPosition(x = x_pos, y = y_pos, z = z_pos)
+
+    @real_position_argument
+    def inverse(self, real_pos):
+        return self.PseudoPosition(calib_z = self.z.position)
+
+    def align(self,z_position=None):
```

Comment:
I was about to check what this used to be called, but the servers are down so I can't. We'll sync the name with whatever the old code defined.

Reply:
Also, the style checker will throw a fit here without a space after the comma:
`def align(self, z_position=None):`